core/src/main/java/org/apache/calcite/sql/SqlCaseN.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+/**
+ * Parse tree for {@code SqlCaseN} expression.
+ */
+public class SqlCaseN extends SqlCall {
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("CASE_N ", SqlKind.OTHER_FUNCTION);
+
+  public final SqlNodeList caseList;
+  public final SqlPartitionByNoneUnknown extraPartitions;
+
+  /**
+   * Creates a {@code SqlCaseN}.
+   * @param pos             Parser position, must not be null
+   * @param caseList           Case expressions
```

Comment:
nit: indentation on "Case expressions"

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/212/commits/596f9761b3fc725a5599e09f123cae7aafa6d4a4